web_responsive/static/src/js/web_responsive.js

```diff
@@ -0,0 +1,238 @@
+/* Copyright 2018 Tecnativa - Jairo Llopis
+ * License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl). */
+
+odoo.define('web_responsive', function (require) {
+    'use strict';
+
+    var AppsMenu = require("web.AppsMenu");
+    var config = require("web.config");
+    var core = require("web.core");
+    var Menu = require("web.Menu");
+
+    /**
+     * Recursively search for menus with their full names
+     *
+     * @param {Object} menu
+     * @returns {Array} It contains many sublevel; you should flatten it
+     */
+    function findNames (menu) {
```

Comment:
Can you put this into the class or return it in web_responsive to expose the function.

Reply:
It is not needed because this is just a helper to generate `this._searchableMenus`, which is the same data that we already had, but in a format usable by `fuzzy.js`. I'll fix docstrings to make this clearer.

If you need to override something, it should be usually `AppsMenu.init()`, where this is used.